### PR TITLE
feat(overthebox.details): add version, system version, last seen, IP service status and reverse DNS

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/otb-device-status`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-266
| License          | BSD 3-Clause

## Description

Add on OverTheBox details :
* device version
* system version
* last time the device has been seen on the provisionning servers
* public IP service status
* reverse DNS for public IP

### CODE REVIEW

File: packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js [hunk 1]
```diff
@@ -462,6 +475,87 @@ export default class OverTheBoxDetailsCtrl {
       });
   }
 
+  /**
+   * Get last seen values
+   */
+  getLastSeen() {
+    if (this.device && this.device.lastSeen) {
```
Comment (JeremyDec) [inline packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js:482]: This test should be done outside of the method. (You can even merge the first test with one at L462)

File: packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js [hunk 2]
```diff
@@ -462,6 +475,87 @@ export default class OverTheBoxDetailsCtrl {
...
+      this.lastSeenAccess = {
+        lastSeen: diffDate,
+        lastSeenHuman: moment.duration(currentDate.diff(lastSeen)).humanize(),
+      };
+
+      if (diffDate < this.OVERTHEBOX_DETAILS.lastSeen.limit) {
+        this.lastSeenAccess.isRecent = true;
+      } else {
+        this.lastSeenAccess.isRecent = false;
+      }
```
Comment (JeremyDec) [inline packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js:497]:
```suggestion
      this.lastSeenAccess.isRecent = diffDate < this.OVERTHEBOX_DETAILS.lastSeen.limit;
```

File: packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js [hunk 3]
```diff
@@ -462,6 +475,87 @@ export default class OverTheBoxDetailsCtrl {
...
+          } else {
+            this.serviceIP = {
+              status: this.OVERTHEBOX_DETAILS.serviceIpStatus.warning,
+            };
+          }
+        } else {
+          this.serviceIP = {
+            status: this.OVERTHEBOX_DETAILS.serviceIpStatus.unknown,
+          };
+        }
```
Comment (JeremyDec) [inline packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js:531]:
```suggestion
        if (!ip || !this.device || !this.device.publicIp) {
          this.serviceIP = {
            status: this.OVERTHEBOX_DETAILS.serviceIpStatus.unknown
          }
        } else {
          this.serviceIP = {
            status: this.device && ip.includes(this.device.publicIp)
              ? this.OVERTHEBOX_DETAILS.serviceIpStatus.locked
              : this.OVERTHEBOX_DETAILS.serviceIpStatus.warning
          }
        }
```

File: packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js [hunk 4]
```diff
@@ -462,6 +475,87 @@ export default class OverTheBoxDetailsCtrl {
...
+        return ips;
+      })
+      .catch((error) => {
+        this.error.checking = error.data;
+        this.TucToast.error(
+          [
+            this.$translate.instant('an_error_occured'),
+            error.data.message,
+          ].join(' '),
```
Comment (JeremyDec) [inline packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js:540]: Once again, you should use string interpolation

File: packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js [hunk 5]
```diff
@@ -462,6 +475,87 @@ export default class OverTheBoxDetailsCtrl {
...
+   * Retrieve reverse DNS from public IP
+   */
+  getReverseDNS() {
+    return this.OvhApiIpReverse.v6()
+      .getReverseDns(this.device.publicIp)
+      .then((dns) => {
+        set(this.device, 'reverse', dns);
+        return dns;
+      })
+      .catch(() => this.$q.when(null));
```
Comment (JeremyDec) [inline packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js:556]:
```suggestion
      .catch(() => null);
```

File: packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js [hunk 6]
```diff
@@ -462,6 +475,87 @@ export default class OverTheBoxDetailsCtrl {
...
+   * Check public IP to retrieve its status
+   */
+  checkPublicIP() {
+    return this.OvhApiIp.v6()
+      .query({
+        'routedTo.serviceName': this.serviceName,
+        type: 'overthebox',
+      })
+      .$promise.then((ips) => {
+        const [ip] = ips;
```
Comment (marie-j) [inline packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js:511]: If `ips` is not needed, I think we can 
```suggestion
      .$promise.then(([ip]) => {
```

File: packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js (hunk 4 above)
Comment (Steffy29) [reply packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js:540]: like that 
this.TucToast.error(
  `${this.$translate.instant('an_error_occured')} ${
   error.data.message}`,
);

Comment (JeremyDec) [reply packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js:540]: Yes :) (Don't forget backticks `)

Comment (Steffy29) [reply packages/manager/modules/overthebox/src/overthebox/details/overTheBox-details.controller.js]: like that 
this.TucToast.error(
  `${this.$translate.instant('an_error_occured')} ${
   error.data.message}`,
);

**Comment by ovh-cds** [comment]
CDS Report installation#14322.0 ✘  
* Stage 1
  * Install ✘

**Review by frenautvh — DISMISSED** [review]

**Review by JeremyDec — DISMISSED** [review]
@Steffy29, your branch needs to be rebased on origin develop

**Comment by antleblanc** [comment]
> @Steffy29, your branch needs to be rebased on origin develop

I'm rebasing this branch on `develop` in order to be able to merge it.

**Comment by antleblanc** [comment]
Rebased successfully. Waiting feedback from CI/CD pipelines. Pull Request got 2 approvals before rebasing.

**Pull request merged**